xcm/pallet-xcm-benchmarks/src/generic/benchmarking.rs

```diff
@@ -0,0 +1,316 @@
+		let instruction = Instruction::ClearOrigin;
+		let xcm = Xcm(vec![instruction]);
+	} : {
+		executor.execute(xcm)?;
+	} verify {
+		assert_eq!(executor.origin, None);
+	}
+
+	report_error {
+		let mut executor = new_executor::<T>(Default::default());
+		executor.error = Some((0u32, XcmError::Unimplemented));
+		let query_id = Default::default();
+		let dest = T::valid_destination().map_err(|_| BenchmarkError::Skip)?;
+		let max_response_weight = Default::default();
+
+		let instruction = Instruction::ReportError { query_id, dest, max_response_weight };
+		let xcm = Xcm(vec![instruction]);
+	}: {
+		executor.execute(xcm)?;
+	} verify {
+		// the execution succeeding is all we need to verify this xcm was successful
+	}
+
+	claim_asset {
+		let (origin, ticket, assets) = T::claimable_asset()
+			.ok_or(BenchmarkError::Skip)?;
+
+		// We place some items into the asset trap to claim.
+		let mut executor = new_executor::<T>(origin.clone());
+		executor.holding = assets.clone().into();
+		executor.execute(Xcm(vec![]))?;
+		executor.post_execute(0);
+		// Assets should be in the trap now.
+
+		let mut executor = new_executor::<T>(origin);
+		let instruction = Instruction::ClaimAsset { assets: assets.clone(), ticket };
+		let xcm = Xcm(vec![instruction]);
+	} :{
+		executor.execute(xcm)?;
+	} verify {
+		assert!(executor.holding.ensure_contains(&assets).is_ok());
+	}
+
+	trap {
+		let mut executor = new_executor::<T>(Default::default());
+		let instruction = Instruction::Trap(10);
+		let xcm = Xcm(vec![instruction]);
+		// In order to access result in the verification below, it needs to be defined here.
+		let mut _result = Ok(());
+	} : {
+		_result = executor.execute(xcm);
+	} verify {
+		match _result {
+			Err(error) if error.xcm_error == XcmError::Trap(10) => {
+				// This is the success condition
+			},
+			_ => panic!("xcm trap did not return the expected error")
+		};
+	}
+
+	subscribe_version {
+		use xcm_executor::traits::VersionChangeNotifier;
+		let origin = T::transact_origin().ok_or(BenchmarkError::Skip)?;
+		let query_id = Default::default();
+		let max_response_weight = Default::default();
+		let mut executor = new_executor::<T>(origin.clone());
+		let instruction = Instruction::SubscribeVersion { query_id, max_response_weight };
+		let xcm = Xcm(vec![instruction]);
+	} : {
+		executor.execute(xcm)?;
+	} verify {
+		assert!(<T::XcmConfig as xcm_executor::Config>::SubscriptionService::is_subscribed(&origin));
+	}
+
+	unsubscribe_version {
+		use xcm_executor::traits::VersionChangeNotifier;
+		// First we need to subscribe to notifications.
+		let origin = T::transact_origin().ok_or(BenchmarkError::Skip)?;
+		let query_id = Default::default();
+		let max_response_weight = Default::default();
+		let mut executor = new_executor::<T>(origin.clone());
+		let instruction = Instruction::SubscribeVersion { query_id, max_response_weight };
+		let xcm = Xcm(vec![instruction]);
+		executor.execute(xcm)?;
```

Comment:
weaker case than above, but you might still consider using `SubscriptionService::start(&origin, query_id, max_response_weight)` directly

Reply:
done!